planner/core/rule_inject_extra_projection.go

```diff
@@ -57,12 +56,6 @@ func (pe *projInjector) inject(plan PhysicalPlan) PhysicalPlan {
 		plan = InjectProjBelowSort(p, p.ByItems)
 	case *NominalSort:
 		plan = TurnNominalSortIntoProj(p, p.OnlyColumn, p.ByItems)
-	case *PhysicalTableReader:
-		tableReader, isTableReader := plan.(*PhysicalTableReader)
-		if isTableReader && tableReader.StoreType == kv.TiFlash {
-			tableReader.tablePlan = pe.inject(tableReader.tablePlan)
-			tableReader.TablePlans = flattenPushDownPlan(tableReader.tablePlan)
-		}
```

Comment:
I suggest keep this because in my understanding, this is the behavior for TiDB root task, so by `inject proj`, it can avoid some compatibility issues, for example: `sum(int)` in TiDB is actually `sum(cast(int as decimal))`, but in TiFlash, it is `cast sum(int) as decimal`, which is not exactly the same as TiDB, by `inject proj`, TiFlash will execute the exactly same expression as in TiDB.

Reply:
let it as a TODO, because injecting projections should keep themself can be pushed down.